neuro_san/internals/interfaces/agent_storage_source.py

```diff
@@ -0,0 +1,25 @@
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from neuro_san.internals.interfaces.agent_network_provider import AgentNetworkProvider
+
+
+class AgentStorageSource:
+    """
+    Interface onto AgentNetworkStorage so that there is not a tangle with the service side.
+    """
+
+    def get_agent_network_provider(self, agent_name: str) -> AgentNetworkProvider:
+        """
+        Get AgentNetworkProvider for a specific agent
+        :param agent_name: name of an agent
+        """
```

Comment:
So far the event handlers that use the source only ever use this one method so far.
This is the basis of the abstraction.
We can always add more methods later if needed.